src/fs/filesystem.ts

```diff
@@ -128,34 +127,41 @@ export class FileSystem {
     const { keyName = 'filesystem-root', rootDid = '' } = opts
 
     const root = await BareTree.fromCID(cid)
-    const publicCID = root.findLinkCID('public')
+    const publicCID = root.links['public']?.cid || null
     const publicTree = publicCID !== null
-      ? await PublicTree.fromCID(publicCID, null)
+      ? await PublicTree.fromCID(publicCID)
       : null
 
     const prettyTree = (await root.getDirectChild('pretty')) as BareTree ||
-      await BareTree.empty()
-    const pinsTree = (await root.getDirectChild('pins')) as BareTree ||
-      await BareTree.empty()
+                        await BareTree.empty()
 
-    const privateCID = root.findLinkCID('private')
+    const privateCID = root.links['private']?.cid || null
     const key = await keystore.getKeyByName(keyName)
     const privateTree = privateCID !== null
       ? await PrivateTree.fromCID(privateCID, key)
       : null
 
     if (publicTree === null || privateTree === null) return null
 
-    return new FileSystem({
+    const fs = new FileSystem({
       root,
       publicTree,
       prettyTree,
       privateTree,
-      pinsTree,
       rootDid
     })
+
+    publicTree.onUpdate = result => fs.updateRootLink(Branch.Public, result)
+    prettyTree.onUpdate = result => fs.updateRootLink(Branch.Pretty, result)
+    privateTree.onUpdate = result => fs.updateRootLink(Branch.Private, result)
+
+    return fs
   }
 
+  static async forUser(username: string, opts: FileSystemOptions = {}): Promise<FileSystem | null> {
+    const cid = await dataRoot.lookup(username)
+    return cid ? FileSystem.fromCID(cid, opts) : null
+  }
```

Comment:
I think this sneaked back in with the rebase?
We have `src/loadFileSystem.ts` now that replaces this.

Reply:
oops thanks for pointing that out! :ok_hand: